### PR TITLE
Mocking of APIs in retrofit, Canceling API update.

### PR DESCRIPTION
**Pull request by DearDhruv**
Added mocking of the APIs.
Updated gradles.
Updated canceling call from retrofit.
And other minor updates.

### CODE REVIEW

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/build.gradle [hunk 1]
```diff
@@ -19,39 +22,65 @@ android {
...
-    annotationProcessor 'com.jakewharton:butterknife-compiler:8.5.1'
-
-    compile('com.mikepenz:aboutlibraries:5.9.4@aar') {
+    implementation supportDependencies.supportV13
+    //
+    implementation supportDependencies.supportVectorDrawable
+    //
+    implementation supportDependencies.exifInterface
+    //
+    implementation 'com.github.bumptech.glide:glide:3.8.0'
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/build.gradle]: You have taken dependencies.gradle separately so I think these all dependencies should also be referenced from that gradle file itself.

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/build.gradle [hunk 2]
```diff
@@ -19,39 +22,65 @@ android {
...
+    implementation 'com.google.code.gson:gson:2.8.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.8.1'
+    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/build.gradle]: either follow dependencies.gradle file way or follow artifact ID version declaring approach.

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/build.gradle [hunk 3]
```diff
@@ -19,39 +22,65 @@ android {
...
+    implementation 'com.squareup.retrofit2:retrofit:2.3.0'
+    implementation 'org.greenrobot:eventbus:3.0.0'
+    implementation 'com.airbnb.android:lottie:2.0.0'
+    implementation 'com.jakewharton:butterknife:8.7.0'
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/build.gradle]: same for butterknife artifact ID version

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -2,13 +2,12 @@
...
           package="com.demo.retrofit">
 
-    <uses-permission android:name="android.permission.INTERNET"/>
-    <uses-permission android:name="android.permission.CAMERA"/>
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
-
+    <uses-permission android:name="android.permission.CAMERA"/>
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/AndroidManifest.xml]: Such formatting only changes should be avoided in commit!

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java [hunk 5]
```diff
@@ -293,14 +280,19 @@ private boolean updatePermissions() {
                     new PermissionResult() {
                         @Override
                         public void permissionGranted() {
+                            startCamera();
                         }
 
                         @Override
                         public void permissionDenied() {
+                            showSnackBar(imgUpload.getRootView(),
+                                    "Please allow permission to take Image.");
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java]: Messages should be declared in strings.xml

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java [hunk 6]
```diff
@@ -235,7 +222,7 @@ private void handleGalleryPicResult(int resultCode, Intent resultData) {
                 addImages(resultData.getData());
             } else {
                 if (resultData.getClipData() != null &&
-                    resultData.getClipData().getItemCount() > 0) {
+                        resultData.getClipData().getItemCount() > 0) {
                     for (int i = 0; i < resultData.getClipData().getItemCount(); i++) {
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java]: how about for_each loop?

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java [hunk 7]
```diff
@@ -293,14 +280,19 @@ private boolean updatePermissions() {
...
                         @Override
                         public void permissionDenied() {
+                            showSnackBar(imgUpload.getRootView(),
+                                    "Please allow permission to take Image.");
                         }
 
                         @Override
                         public void permissionForeverDenied() {
+                            showSnackBar(imgUpload.getRootView(),
+                                    "Please allow permission from Application setting.");
```
Comment (PareshMayani) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java]: same comment of declaring messages in strings.xml file

File: android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java (hunk 6 above)
Comment (DearDhruv) [inline android/examples_demos/Retrofit-with-EventBus/Retrofit_With_EventBus/app/src/main/java/com/demo/retrofit/fragments/UploadFragment.java]: foreach not applicable to type 
> android.content.ClipData

**Comment by DearDhruv** [comment]
All changes reflected. 
Although for `ClipData`  foreach loop is not applicable .

**Comment by PareshMayani** [comment]
Thanks @DearDhruv for building an example and sharing.

**Comment by DearDhruv** [comment]
Thanks, @PareshMayani 
I shall make the example more richer as time permits.

**Pull request merged**